flocker/node/script.py

```diff
@@ -179,11 +132,47 @@ def get_service(self, reactor, options):
         """
         return AgentLoopService(
             reactor=reactor,
-            deployer=self.deployer_factory(hostname=options["hostname"]),
+            deployer=self.deployer_factory(options=options),
             host=options["destination-host"], port=options["destination-port"],
         )
 
 
+def volume_service_from_config(config):
+    pass
+
+def dataset_deployer_from_options(options):
```

Comment:
As mentioned elsewhere, the signature of this method should be different.  Perhaps something more like `dataset_deployer_from_options(backend_name, backend_configuration)`